### PR TITLE
feat: add support for azure openai

### PR DESCRIPTION
**Pull request by cpendery**
Adds support for models hosted on Azure OpenAI under the model prefix `azure:`. 

Part of #14 & #22

### CODE REVIEW

**Comment by cpendery** [comment]
> Thanks for opening this PR.
> 
> This seems to introduce a lot of code duplication. Couldn't you just subclass `OpenAIModel`? Or perhaps even better to add some logic in `OpenAIModel.__init__` to just initialize `self.client` according to the prefix of `SHORTCUTS`?

I’m happy to do either, I was just trying to avoid changing the convention of one class extending the base per model provider. I’ll go with the updating the OpenAIModel

**Review by klieret — APPROVED** [review]
LGTM! If you confirm that you've tested it, I think we're good to merge :)

**Comment by cpendery** [comment]
> LGTM! If you confirm that you've tested it, I think we're good to merge :)

If someone else can validate it before we merge, that would be great. I'm only on Windows so all the other issues are blocking me for verifying this works for the whole pipeline

**Comment by klieret** [comment]
Let me ping @0xdevalias and @kim-borgen since they upvoted. Perhaps they can test this.

**Comment by 0xdevalias** [comment]
> since they upvoted. Perhaps they can test this.

@klieret I'm not currently setup to be able to test this unfortunately. I just liked the idea of wider support.

**Comment by kim-borgen** [comment]
@klieret Unfortunately I only have the 8k context length available for testing now and the first prompt (messages) is 10,963. Other than that, it seems to work. Sorry but I can't dedicate more time, but if someone supplies a config that uses less than 6k tokens for example I can test quickly 

Otherwise: 
- Requested change: add api_version to cfg keys
- When trying to get the API working I get this error, so I had to delete the traj/kim folder every time, a bit annoying. 
```
WARNING  **************************************************                                                                                                                                                                                                                                                                                                                                                              
WARNING  Found existing args.yaml with different arguments!                                                                                                                                                                                                                                                                                                                                                              
WARNING  **************************************************   
```

**Comment by klieret** [comment]
@kim-borgen Thanks for the quick check!  Can you open issues with some more context for the "Otherwise" ;)  (not sure if I fully understand the context of the second one yet)

File: sweagent/agent/models.py
Comment (mspronesti) [inline sweagent/agent/models.py:228]: How about parametrizing the `api_version` too? It changes rather frequently :)
```suggestion
             self.api_model = cfg["AZURE_OPENAI_DEPLOYMENT"]
             api_version = cfg["OPENAI_API_VERSION"] or "2024-02-01"
             self.client = AzureOpenAI(api_key=cfg["AZURE_OPENAI_API_KEY"], azure_endpoint=cfg["AZURE_OPENAI_ENDPOINT"], api_version=api_version)
```

**Review by mspronesti — CHANGES_REQUESTED** [review]

File: sweagent/agent/models.py
Comment (klieret) [reply sweagent/agent/models.py:228]: we could make it a `.get("...", default value)` to get the best of both worlds

Comment (mspronesti) [reply sweagent/agent/models.py:228]: Suggestion updated :)

Comment (klieret) [reply sweagent/agent/models.py:228]: Thanks! Could you check that this works? (I don't have azure credits atm)

**Comment by klieret** [comment]
Since this doesn't seem to impact any other models anyway, I've gone ahead and merged it :)

**Comment by klieret** [comment]
Note: We shouldn't override the `--model_name`, so I've removed `AZURE_OPENAI_DEPLOYMENT` in #127

**Pull request merged**
